src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -775,6 +775,13 @@ private bool ReadSingleSegment()
             bool retVal = false;
             ValueSpan = default;
 
+            // Handle BOM if present
+            int utf8BomLength = JsonConstants.Utf8Bom.Length;
+            if (_buffer.Length >= utf8BomLength && JsonConstants.Utf8Bom.SequenceEqual(_buffer.Slice(0, utf8BomLength)))
+            {
+                _consumed += JsonConstants.Utf8Bom.Length;
+            }
```

Comment:
Can this code be refactored to use only version of it, as it is an exact copy of the above code?